### PR TITLE
[mlir][linalg] Move transpose_matmul to targeted transform op

### PR DESCRIPTION
**Pull request by c-rhodes**
More targeted than a blanket "apply everywhere" pattern. Follow up to #89075 to address @ftynse's feedback.

### CODE REVIEW

File: mlir/lib/Dialect/Linalg/TransformOps/LinalgTransformOps.cpp [hunk 1]
```diff
@@ -3422,6 +3416,34 @@ DiagnosedSilenceableFailure transform::TransposeConv2DOp::applyToOne(
...
+            return transposeMatmul(rewriter, op, transposeLHS);
+          })
+          .Case([&](linalg::BatchMatmulOp op) {
+            return transposeBatchMatmul(rewriter, op, transposeLHS);
+          })
+          .Default([&](Operation *op) {
+            return rewriter.notifyMatchFailure(op, "not supported");
+          });
+  if (failed(maybeTransformed))
+    return emitDefaultSilenceableFailure(target);
```
Comment (ftynse) [inline mlir/lib/Dialect/Linalg/TransformOps/LinalgTransformOps.cpp:3441]: Nit: could we rather say "not supported" here than in the `notifyMatchFailure` that is ignored by the infra?

**Review by ftynse — APPROVED** [review]
Thanks! I assume most of the PR just moves the code around so I didn't review it in detail. LMK if attention is needed somewhere.

File: mlir/lib/Dialect/Linalg/TransformOps/LinalgTransformOps.cpp (hunk 1 above)
Comment (c-rhodes) [reply mlir/lib/Dialect/Linalg/TransformOps/LinalgTransformOps.cpp]: I've updated it to simply return failure from the TypeSwitch for unsupported op and emit a message here instead.

**Comment by c-rhodes** [comment]
> Thanks! I assume most of the PR just moves the code around so I didn't review it in detail. LMK if attention is needed somewhere.

Thanks for speedy review! It's mostly mechanical, no changes to the actual rewrites.

**Pull request merged**